tools/crates/cli/src/commands/starknet/transfer_args.rs

```diff
@@ -0,0 +1,150 @@
+use cgp::prelude::*;
+use hermes_cli_components::traits::build::CanLoadBuilder;
+use hermes_cli_components::traits::command::CommandRunner;
+use hermes_cli_components::traits::output::HasOutputType;
+use hermes_encoding_components::traits::encode::CanEncode;
+use hermes_logging_components::traits::has_logger::HasLogger;
+use hermes_logging_components::traits::logger::CanLog;
+use hermes_logging_components::types::level::LevelInfo;
+use hermes_relayer_components::chain::traits::queries::chain_status::CanQueryChainStatus;
+use hermes_starknet_chain_components::types::channel_id::ChannelId;
+use hermes_starknet_chain_components::types::cosmos::height::Height;
+use hermes_starknet_chain_components::types::cosmos::timestamp::Timestamp;
+use hermes_starknet_chain_components::types::messages::ibc::channel::PortId;
+use hermes_starknet_chain_components::types::messages::ibc::denom::{
+    Denom, PrefixedDenom, TracePrefix,
+};
+use hermes_starknet_chain_components::types::messages::ibc::ibc_transfer::MsgTransfer;
+use hermes_starknet_chain_context::contexts::encoding::cairo::StarknetCairoEncoding;
+use ibc::core::host::types::identifiers::PortId as IbcPortId;
+use starknet::core::types::Felt;
+
+use crate::contexts::app::ToolApp;
+
+#[derive(Debug, clap::Parser, HasField)]
+pub struct TransferArgs {
+    /// Amount to transfer
+    #[clap(
+        long = "amount",
+        required = true,
+        value_name = "AMOUNT",
+        help_heading = "REQUIRED"
+    )]
+    pub amount: String,
+
+    /// Denom of the amount to transfer
+    #[clap(
+        long = "denom",
+        required = true,
+        value_name = "DENOM",
+        help_heading = "REQUIRED"
+    )]
+    pub denom: String,
+
+    /// Address of the receiver
+    #[clap(
+        long = "receiver",
+        required = true,
+        value_name = "RECEIVER",
+        help_heading = "REQUIRED"
+    )]
+    pub receiver: String,
+
+    /// Address of the sender
+    #[clap(
+        long = "sender",
+        required = true,
+        value_name = "SENDER",
+        help_heading = "REQUIRED"
+    )]
+    pub sender: String,
+
+    /// ID of the channel used for the transfer
+    #[clap(
+        long = "channel-id",
+        required = true,
+        value_name = "CHANNEL_ID",
+        help_heading = "REQUIRED"
+    )]
+    pub channel_id: String,
+}
+
+pub struct RunTransferArgs;
+
+#[async_trait]
+impl CommandRunner<ToolApp, TransferArgs> for RunTransferArgs {
+    async fn run_command(
+        app: &ToolApp,
+        args: &TransferArgs,
+    ) -> Result<<ToolApp as HasOutputType>::Output, <ToolApp as HasErrorType>::Error> {
+        let builder = app.load_builder().await?;
```

Comment:
This line attempts to load a `config.toml` file while loading the app, and it's not clear where that file should live. If that's necessary for this CLI tool, I suggest the CLI either be taking the file as an argument or have the required parameters hardcoded for better independent control over them.

Reply:
Instead of using the chain_status to set the timeout, the user can now specify the explicit timeout. This allows us to remove the need to load the builder and other dependencies which made the UX worse. Done here 96ff0dafb62b61622c2ffdc26ff71f90c0caf234